packages/terra-paginator/src/ProgressivePaginator.jsx

```diff
@@ -230,63 +230,69 @@ class ProgressivePaginator extends React.Component {
-            <span className={cx('icon')} />
-          </PaginatorButton>
         </div>
-        <div>
+        <ul className={cx('progressive-list')}>
           {
-            intl.formatMessage({ id: messageId }, messageAttributes)
+            (totalCount) && (
+              <li>
+                <PaginatorButton
+                  ariaDisabled={selectedPage === 1}
+                  ariaLabel={intl.formatMessage({ id: 'Terra.paginator.first' })}
+                  className={cx(['nav-link', selectedPage === 1 ? 'is-disabled' : null])}
+                  disabled={selectedPage === 1}
+                  onClick={this.handlePageChange(1)}
+                  onKeyDown={this.handlePageChange(1)}
+                >
+                  {intl.formatMessage({ id: 'Terra.paginator.first' })}
+                </PaginatorButton>
+              </li>
+            )
           }
-        </div>
-        <div>
-          <PaginatorButton
-            ariaDisabled={selectedPage === totalPages}
-            ariaLabel={intl.formatMessage({ id: 'Terra.paginator.next' })}
-            className={cx(['nav-link', 'next', 'icon-only', selectedPage === totalPages ? 'is-disabled' : null])}
-            disabled={selectedPage === totalPages}
-            onClick={this.handlePageChange(nextPageIndex)}
-            onKeyDown={this.handlePageChange(nextPageIndex)}
-          >
-            <VisuallyHiddenText text={intl.formatMessage({ id: 'Terra.paginator.next' })} />
-            <span className={cx('icon')} />
-          </PaginatorButton>
+          <li>
+            <PaginatorButton
+              ariaDisabled={selectedPage === 1}
+              ariaLabel={intl.formatMessage({ id: 'Terra.paginator.previous' })}
+              className={cx(['nav-link', 'previous', 'icon-only', selectedPage === 1 ? 'is-disabled' : null])}
+              disabled={selectedPage === 1}
+              onClick={this.handlePageChange(previousPageIndex)}
+              onKeyDown={this.handlePageChange(previousPageIndex)}
+            >
+              <VisuallyHiddenText text={intl.formatMessage({ id: 'Terra.paginator.previous' })} />
+              <span className={cx('icon')} />
+            </PaginatorButton>
+          </li>
+          <li>
+            <PaginatorButton
+              ariaDisabled={selectedPage === totalPages}
+              ariaLabel={intl.formatMessage({ id: 'Terra.paginator.next' })}
+              className={cx(['nav-link', 'next', 'icon-only', selectedPage === totalPages ? 'is-disabled' : null])}
+              disabled={selectedPage === totalPages}
+              onClick={this.handlePageChange(nextPageIndex)}
+              onKeyDown={this.handlePageChange(nextPageIndex)}
+            >
+              <VisuallyHiddenText text={intl.formatMessage({ id: 'Terra.paginator.next' })} />
+              <span className={cx('icon')} />
+            </PaginatorButton>
+          </li>
           {
             (totalCount) && (
```

Comment:
Do we need to have another one bracket and the `totalCount` here? Can we combine with the one above?

Reply:
This one is for the "Last" button and another one is for the "First" button. We could potentially have something like:
```
const buttonsList = totalCount ?
(
  <FirstButton>
  <PreviousButton>
  <NextButton>
  <LastButton>
) 
: (
  <PreviousButton>
  <NextButton>
)
```
which will duplicate the PreviousButton and NextButton codes, so I just prefer to have both _totalCount_ to make it simple. Thoughts?

Reply:
Oh, I see it now. The `totalCount` value is used to control whether to render the first and last button based on whether the total count is determined or not. Perhaps it's clearer to use a boolean variable name?

```
renderLastButton = totalCount > 0 ? true : false;
```